### PR TITLE
[ci skip] Update active_record_basics table naming section

### PR DESCRIPTION
**Pull request by chase-dougherty**
I found that the naming conventions did not explain how modules are taken into account, but Active Record [compute_table_name](https://github.com/rails/rails/blob/b943622bdc746370ac860bfd3240cc0b8ca59d90/activerecord/lib/active_record/model_schema.rb#L606) did. So I added an explanation into the documentation so future devs understand how modules are being used.

About this template

### Motivation / Background

I was recently working with a new model that was to be nested under multiple modules. The naming convention proved confusing as the documentation did not mention the need for modules to be taken into account, and let alone it wasn't taking all the modules into account. I found that it only takes the first module parent into account, and only when it's not an `abstract_class`.

For example

```ruby
module Library
  module Bookshelf
    class Book < ApplicationRecord
    end
  end
end
```
The above would produce `bookshelf_books` and not `library_bookshelf_books`

This Pull Request has been created because I want to prevent others from having to go deep into Active Record code to figure out the logic here.

### Detail

This Pull Request changes the active record naming conventions section to include modules.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: guides/source/active_record_basics.md [hunk 1]
```diff
@@ -125,6 +125,12 @@ Here are some more examples of model class names and corresponding table names:
 | `Product`        | `products`     |
 | `Person`         | `people`       |
 
+For models nested inside of a module, only the first parent module will be used as
+a prefix onto the table. It will first be singularized, and then followed by an underscore.
+
+If the module is labeled as `abstract`, then it will
+not be used as a prefix.
+
```
Comment (Venkat-RK) [inline guides/source/active_record_basics.md:133]: Hi, 
Thanks for adding additional notes for better understanding. 
Could you please add an example code snippet for better readability?

Comment (chase-dougherty) [reply guides/source/active_record_basics.md:133]: Done!

Comment (Venkat-RK) [reply guides/source/active_record_basics.md:133]: Thank you!

**Review by Venkat-RK — COMMENTED** [review]
Nice write up!

**Comment by p8** [comment]
Thanks @chase-dougherty !
Can you squash your commits?

**Comment by chase-dougherty** [comment]
@p8 thank you! All done!

**Comment by zzak** [comment]
I'm not sure this makes sense so early in the AR Basics guide, it's something you might run into later on, but definitely not someone who is just learning about AR for the first time.

**Comment by chase-dougherty** [comment]
@zzak I see what you are saying because this is a "Basic" section and this piece of information doesn't seem "basic". 

Where this is stemming from is that we have a lot of people new to Rails at our work and they were going here to figure out how to name our tables. As you know, a lot of our models are nested deeply because of our codebase size. Most companies I have worked at have their models nested under a single module though. Shopify is the first place I have been at with multiple nested modules and we are manually setting our table names on our models because the Active Record naming magic is hard to find and understand. 

A couple of ideas

1. We could change this information to talk about a single module and how it's attached. Then end with a small note that multiple nested modules are not included. This would make the information more basic and it would fit in more with this section.
2. We could append a link at the end of this that redirects them to a more advanced section? Do we have a more advanced section for table naming?

**Comment by p8** [comment]
@chase-dougherty I think expanding the API documentation of `compute_table_name` would help. 
We can link to that form the guides for more advanced examples.

**Comment by chase-dougherty** [comment]
Closing this PR since using modules with models was deemed outside of the basic use case. If we update the API documentation of `compute_table_name`, then we can consider adding a link here redirecting readers to there for more learning.

**Pull request closed without merging**